tests/helpers.js

```diff
@@ -0,0 +1,58 @@
+const { MongoClient, Collection } = require('mongodb');
+const { createApolloFetch } = require('apollo-fetch');
+const { Admin, User, ADMIN_TOKEN } = require('./seeds/users');
+
+Collection.prototype.findOrInsertOne = async function findOrInsertOne(
+  doc,
+  ...args
+) {
+  try {
+    const { insertedId } = await this.insertOne(doc, ...args);
+    return this.findOne({ _id: insertedId }, ...args);
+  } catch (e) {
+    return this.findOne({ _id: doc._id }, ...args);
+  }
```

Comment:
it is generally advisable to not do too much magic, maybe its better to be precice in tests on insert or find

Reply:
I had race conditions because the top level test suits all tried to create the admin user at the same time, i fixed it by turning things around:

before:
await findOne -> conditionally await insert

now:
try await insert -> await findOne even on catch

You're definitively right, this is a little bit of magic. Maybe I tried to solve things the wrong way, how could I prepare the db for the tests in a more safe way?

Reply:
There is --maxConcurrency=